ticker/src/main/java/com/robinhood/ticker/TickerColumn.java

```diff
@@ -136,10 +136,35 @@ private void setCharacterIndices() {
                 currentCharacterList = new char[] { currentChar };
                 startIndex = endIndex = 0;
             } else {
-                currentCharacterList = new char[]{currentChar, targetChar};
+                currentCharacterList = new char[] {currentChar, targetChar};
                 startIndex = 0;
                 endIndex = 1;
             }
+        } else if (wraparound && endIndex < startIndex) {
+            // Modify the current character list & indices to take wraparound into account
+            // NOTE(jin): Some hackery here to avoid copying over the empty character sigh
+            // TODO(jin) revisit this hackery for empty character
+            final boolean isEndingOnEmptyChar = endIndex == 0;
+            final int numFromStart = currentCharacterList.length - startIndex;
+            final int numToEnd = endIndex + (isEndingOnEmptyChar ? 1 : 0);
+            final char[] modifiedWraparoundCharList = new char[numFromStart + numToEnd];
+            System.arraycopy(
+                    currentCharacterList,
+                    startIndex,
+                    modifiedWraparoundCharList,
+                    0,
+                    numFromStart
+            );
+            System.arraycopy(
+                    currentCharacterList,
+                    isEndingOnEmptyChar ? 0 : 1,
+                    modifiedWraparoundCharList,
+                    numFromStart,
+                    numToEnd
+            );
+            currentCharacterList = modifiedWraparoundCharList;
```

Comment:
i wonder if some of the internals here would be made a lot easier by using a circular buffer/ring type instead of a list? i'm not sure if it complicates other places, but we could compute which direction would be a shorter distance to travel and use that to determine whether we should go upwards or downwards. i think it would avoid needing to do these arraycopys, and possibly some other shenanigans.

Reply:
yeah i started going down that route, but it ended up being quite a bit of change for a somewhat uncommon use-case (a lot of the internal drawing logic had to be re-worked). I decided that it was easier to just use the exact same drawing logic and just properly set up the constraints for it.

i do think some of this logic can be simplified using better data structure, just too lazy to do it now lol..